flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -140,6 +147,25 @@ object UserDefinedFunctionUtils {
           s"one method named 'eval' which is public, not abstract and " +
           s"(in case of table functions) not static.")
     } else {
+      var trailingSeq = false
+      var trailingArray = false
+      methods.foreach(method => {
+        val signatures = method.getParameterTypes
+        if (signatures.nonEmpty) {
+          val trailingArg = signatures(signatures.length - 1)
+          if (trailingArg.getName.equals("scala.collection.Seq")) {
+            trailingSeq = true
+          } else if (trailingArg.isArray) {
+            trailingArray = true
+          }
+        }
+      })
+      if (trailingSeq && !trailingArray) {
```

Comment:
If the users use the annotation `@scala.annotation.varargs`, Scala will generate two signatures of the method. One is `T eval(scala.collection.Seq<T> args)`, the other is `T eval(T[] args)`. A better idea is to compare every arguments of the signature. We can make sure either there is only one method `T eval(T[] args)`, or there are two methods: `T eval(scala.collection.Seq<T> args)` and `T eval(T[] args)`.